src/main.ts

```diff
@@ -88,6 +92,11 @@ async function run() {
       stack.resolver ?? stackYaml.resolver
     }`;
 
+    const hieExists = fs.existsSync(HIE_YAML);
+    const compilerTools = hieExists
+      ? inputs.compilerTools.concat(["implicit-hie"])
```

Comment:
We probably need further options here. Users may have an `hie.yaml` but know care to use `implicit-hie` and might be using a resolver where it's not available. Such builds would fail without any way for those users to avoid the error besides mucking with `extra-deps` or removing their `hie.yaml`.